### DISCUSSION TITLE
Seeking Help with Keyboard Shortcuts for Column Movement and Auto-Closing Tags in Web Development

### DISCUSSION THREAD

**Discussion by danielR1234** [zed-industries/zed#8198] [Feature Requests]
Hey guys, I have two questions.

First, how do I set up a shortcut so that I can move the columns up and down? Unfortunately, I have no idea what this feature is called, haha. (see video)

Second question, is there something like auto-closing tags? I mean, when I create a <div> is there a way for a </div>
to be automatically generated?

https://github.com/zed-industries/zed/assets/35528731/d1d0c1bc-6ded-419a-a114-f5f626b2a45a

**Comment by Moshyfawn** [comment]
The auto-close tags feature doesn't currently exist in Zed. You can see #6592 for more context.

As for the `MoveLine` shortcut, in Zed it's mapped to `ctrl-cmd-up` and `ctrl-cmd-down` by default.

```json
[
  {
    "context": "Editor",
    "bindings": {
      "ctrl-cmd-up": "editor::MoveLineUp",
      "ctrl-cmd-down": "editor::MoveLineDown"
    }
  }
]

```

**Comment by danielR1234** [comment]
Thank you for your response. I tried to overwrite the settings with it. Unfortunately, it's not possible to save. The IDE completely freezes.

```
"option-up": "editor::MoveLineUp",
"option-down": "editor::MoveLineDown"
```

**Reply by Moshyfawn** [reply]
Have you tried editing the `Default Key Bindings` file or the `~/.config/zed/keymap.json` file?

**Reply by Moshyfawn** [reply]
In addition, the `option` key is called `alt` in Zed.

**Reply by danielR1234** [reply]
Default Key Bindings that correct? :D

**Reply by Moshyfawn** [reply]
`Default Key Bindings` serves as the reference file for the default keymaps configuration. To override any of the keymaps, edit the `~/.config/zed/keymap.json` file instead. You can access it by pressing `cmd-shift-p` to open the command palette and searching for `zed: open keymap`. 

```json
[
  {
    "context": "Editor",
    "bindings": {
      "alt-up": "editor::MoveLineUp",
      "alt-down": "editor::MoveLineDown"
    }
  }
]

```

**Reply by Moshyfawn** [reply]
Note that both `alt-up` and `alt-down` keymaps are bound to select syntax nodes by default.

```json
[
  {
    "context": "Editor",
    "bindings": {
      "alt-up": "editor::SelectLargerSyntaxNode",
      "alt-down": "editor::SelectSmallerSyntaxNode",
    }
  }
]
```

**Reply by danielR1234** [reply]
Thx. Its working now!
